src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +379,168 @@ else if (!period.isEmpty()) {
         }
     }
 
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    public void validateSummaryTag(DetailNode ast) {
+        if (getInlineSummary(ast).isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            final String inlineSummaryDoc = getInlineSummary(ast);
+            if (isPeriodAtEnd(inlineSummaryDoc, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
+            else if (isSummaryValid(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
+            }
+        }
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
```

Comment:
please order method in sequence of how they become used in code.
you can generate matrix of usage by https://github.com/sevntu-checkstyle/methods-distance and you can minimize distances for methods that you created.
webservice is not not available .... there are some prolems, but you can reuse CLI.

Reply:
I tried to use it, but didn't understand perfectly how to place the methods. Please check if current status work.

Reply:
it minimize distance between first usage and declaration of method, so it let to read code easy without jumping and scrolling.

Reply:
Thanks, initially I didn't understood the meaning of positives and negatives and fractional valus, now I get it.